### PR TITLE
feat: provide CUDA image for AMD64

### LINKED ISSUE #36: Add images for CUDA

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #36

### What does this PR do?

Build image for CUDA (AMD64)

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

#36 

### How to test this PR?

N/A

### CODE REVIEW

File: chat/cuda/amd64/Containerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM quay.io/opendatahub/workbench-images:cuda-ubi9-python-3.9-20231206
```
Comment (benoitf) [inline chat/cuda/amd64/Containerfile:1]: should add copyright headers

File: chat/cuda/amd64/Containerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+USER root
+RUN  dnf install -y gcc-toolset-13-gcc gcc-toolset-13-gcc-c++
+USER 1001
+WORKDIR /locallm
+COPY src .
+ENV CMAKE_ARGS="-DGGML_CUDA=on -DLLAMA_AVX2=OFF -DLLAMA_FMA=OFF -DLLAMA_F16C=OFF"
+ENV FORCE_CMAKE=1
+RUN CC="/opt/rh/gcc-toolset-13/root/usr/bin/gcc" CXX="/opt/rh/gcc-toolset-13/root/usr/bin/g++" pip install --no-cache-dir -r ./requirements.txt
+ENTRYPOINT [ "sh", "run.sh" ]
```
Comment (benoitf) [inline chat/cuda/amd64/Containerfile:10]: couldn't it be simplified to only "run.sh" ?

**Review by benoitf — APPROVED** [review]

File: chat/cuda/amd64/Containerfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
+FROM quay.io/opendatahub/workbench-images:cuda-ubi9-python-3.9-20231206
+USER root
+RUN  dnf install -y gcc-toolset-13-gcc gcc-toolset-13-gcc-c++
+USER 1001
+WORKDIR /locallm
+COPY src .
```
Comment (lstocchi) [inline chat/cuda/amd64/Containerfile:6]: GH workflow failed here, src does not exists

File: chat/cuda/amd64/Containerfile (hunk 2 above)
Comment (jeffmaury) [reply chat/cuda/amd64/Containerfile:27]: Followed the same pattern as other Containerfiles from this repo

**Pull request merged**